src/NeoSharp.Core/Network/ServerContext.cs

```diff
@@ -35,30 +38,39 @@ public VersionPayload Version
             }
         }
 
+        /// <inheritdoc />
+        public IEnumerable<EndPoint> ConfigurableEndpoints { get; private set; }
+
+        public NetworkAclConfig NetworkAclConfig { get; private set; }
         #endregion
 
         /// <summary>
         /// Server context
         /// </summary>
-        /// <param name="config">Config</param>
+        /// <param name="networkConfig">Network configuration.</param>
         /// <param name="blockchain">Blockchain</param>
-        public ServerContext(NetworkConfig config, IBlockchain blockchain)
+        public ServerContext(NetworkConfig networkConfig, IBlockchain blockchain)
         {
-            if (config == null) throw new ArgumentNullException(nameof(config));
-            _blockchain = blockchain ?? throw new ArgumentNullException(nameof(blockchain));
+            Guard.ThrowIfNull(networkConfig, nameof(networkConfig));
+            this._blockchain = Guard.ThrowIfNull(blockchain, nameof(blockchain));
 
-            _version = new VersionPayload
+            this._version = new VersionPayload
             {
                 Version = 2,
                 // TODO: What's it?
                 // Services = NetworkAddressWithTime.NODE_NETWORK;
                 Timestamp = DateTime.UtcNow.ToTimestamp(),
-                Port = config.Port,
+                Port = networkConfig.Port,
                 Nonce = (uint)new Random(Environment.TickCount).Next(),
                 UserAgent = $"/NEO:{Assembly.GetExecutingAssembly().GetName().Version.ToString(3)}/",
                 CurrentBlockIndex = _blockchain.CurrentBlock?.Index ?? 0,
                 Relay = true
             };
+
+            // TODO: Change after port forwarding implementation
+            this.ConfigurableEndpoints = networkConfig.PeerEndPoints.ToList();
```

Comment:
No sure, we are going to implement port forwarding here

Reply:
then when??? 
why should be in Server?

Reply:
I think port forwarding has to be solved on a lower level (tcp) and server should adopt this after on configured endpoints.